pages/index.js

```diff
@@ -18,7 +18,7 @@ export async function getStaticProps(context) {
       allFeedback: feedback,
       site
     },
-    unstable_revalidate: 1
+    revalidate: 1
```

Comment:
@leerob Trying to understand SWR vs NextJS incremental static regeneration.

If I understand correctly `revalidate: 1` will effectively make a new Firestore call every second for every active client. In the cases of apps with a lot of traffic, this might affect the Firebase bill quite a bit.

So to avoid the above, one could instead fetch all feedback using `useSWR(user ? ['/api/feedback', user.token])` here which would only make a new Firestore read per active client whenever one of the active SWRs with the same key updated (i.e. either [delete feedback](https://github.com/leerob/fastfeedback/blob/5f021b0b2d9e7deae344d6afe144eb4c2b48b30f/components/DeleteFeedbackButton.js#L23) or [update feedback](https://github.com/leerob/fastfeedback/blob/20b5eabf7bb649f49915375160eb8b5543429e66/components/FeedbackRow.js#L14)).

Does this reasoning make sense to you?

Reply:
> If I understand correctly revalidate: 1 will effectively make a new Firestore call every second for every active client. In the cases of apps with a lot of traffic, this might affect the Firebase bill quite a bit.

Not exactly. It defines a revalidation period of 1 second. If no one viewed your site for a day, it would only make the request to Firebase when the next user visits your site, since 1 second period has passed. It doesn't run _every_ second.

Reply:
Of course, but for every _active client_ will it keep regenerating every second during the time the app is open in someone's browser tab?

Reply:
Nope. Try giving this a read: https://vercel.com/blog/nextjs-server-side-rendering-vs-static-generation#updating-existing-pages-(incremental-static-%22re%22generation)

It generates the page in the background, once, will still serving the previous static page. If it builds successfully, it invalidates the cache and updates with the new static page. Only once, and no downtime.